sled-agent/config-reconciler/src/dataset_serialization_task.rs

```diff
@@ -1965,7 +1927,17 @@ mod illumos_tests {
                     "Attempting automated cleanup of {}",
                     vdev_dir.path(),
                 );
-                self.cleanup();
+
+                // NOTE: There is an assertion in RealZfsTestHarness::new
```

Comment:
Since cleanup is async, this was a little odd to deal with.

The whole point of this pathway is "Dealing with an unclean shutdown", so it's gonna be kinda gnarly regardless (there's no great time to have a caller explicitly call an async fn).

That said, open to feedback if this feels too bad!